spec/physical_field_spec.rb

```diff
@@ -0,0 +1,66 @@
+RSpec.describe SparqlToSwSolr::InstanceSolrDoc::PhysicalField do
+
+  let(:instance_uri) { 'http://ld4p-test.stanford.edu/1234567890#Instance' }
+  let(:isd) { SparqlToSwSolr::InstanceSolrDoc.new(instance_uri) }
+  let(:sparql_conn) { double('sparql client') }
+  let(:solutions) { RDF::Query::Solutions.new }
+
+  before(:each) do
+    allow(isd).to receive(:sparql).and_return(sparql_conn)
+    allow(sparql_conn).to receive(:query).and_return(solutions)
+  end
+
+  context '#physical_values' do
+    it 'is a String consisting of the results from a query solution' do
+      solutions << RDF::Query::Solution.new(noteType: 'Physical details',
+                                            extentLabel_1: '15 p.',
+                                            noteLabel: 'ill.',
+                                            instanceDimensions_3: '5 x 5 in.')
+      physical_values = isd.send(:physical_values)
+      expect(physical_values).to eq '15 p. : ill. ; 5 x 5 in.'
+    end
+    it 'includes noteLabel for "accompanying material" when it exists as a solutions with a different noteType' do
+      solutions << RDF::Query::Solution.new(noteType: 'Physical details',
+                                            extentLabel_1: '15 p.',
+                                            noteLabel: 'ill.',
+                                            instanceDimensions_3: '5 x 5 in.')
+      solutions << RDF::Query::Solution.new(noteType: 'Accompanying materials',
+                                            extentLabel_1: '15 p.',
+                                            noteLabel: 'CD',
+                                            instanceDimensions_3: '5 x 5 in.')
+      physical_values = isd.send(:physical_values)
+      expect(physical_values).to eq '15 p. : ill. ; 5 x 5 in. + CD'
+    end
+    it 'Concatenates the physical_details correctly if there is no extentLabel_1' do
+      solutions << RDF::Query::Solution.new(noteType: 'Physical details',
+                                            extentLabel_1: nil,
```

Comment:
It actually requires the binding, otherwise:
```
 NoMethodError:
       undefined method `extentLabel_1' for #<RDF::Query::Solution:0x007f9ef3289378>
```

I changed the test so that it checks in case the extend of dimensions end up as an empty string, but the sparql query actually ensures that there be a result for each of those bindings.